Занятие 2: Объекты. Задача 1: Функция sumSalary

Detrimon opened the pull request:
Добавлен код функции по подсчету зарплаты.

Detrimon committed:
Занятие 2: Объекты. Задача 1: Функция sumSalary
Добавлен код функции по подсчету зарплаты.

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

Detrimon committed:
Занятие 2: Объекты. Задача 2: Определите, пуст ли Объект
Заходит в цикл, значит не пуст и можно сразу вернуть false (т.е. не пуст). А если true не вернется, значит пойдет
дальше по коду и вернет true (т.е. объект пуст).

jsru commented:
Поздравляю, все задачи решены абсолютно верно
```diff
Функция sumSalary(salaries)
+ должна возвращать 0 если нет свойств с зарплатами
+ должна игнорировать свойства других типов
+ должна складывать все зарплаты
```

Добавляю преподавателя (@stas-dolgachov) для код-ревью, посмотрим что он скажет :)

Detrimon committed:
Merge branch 'master' of https://github.com/js-tasks-ru/jsbasic-20200727_detrimon

jsru commented:
@stas-dolgachov ping :)

Detrimon committed:
Занятие 2: Объекты. Задача 3: Создайте калькулятор
Простая реализация калькулятора с попыткой привести строку в число при помощи оператора +.

dolgachio commented on 2-module/1-task/index.js:
```diff
@@ -5,4 +5,13 @@
+  let sum = 0;
+
+  for (let elem in salaries) {
+    if (isFinite(salaries[elem])) { // или можно использовать typeof salaries[elem] === 'number'. А что предпочтительнее использовать?
```
В этом случае не принципиально, что вам больше нравится :)

dolgachio commented on 2-module/2-task/index.js:
```diff
@@ -5,4 +5,8 @@
  */
 function isEmpty(obj) {
   // ваш код...
+  for (let elem in obj) {
```
👍

dolgachio commented on 2-module/3-task/index.js:
```diff
@@ -1,5 +1,17 @@
 let calculator = {
   // ваш код
+  read(a, b) {
```
👍
